nextflow/main.nf

```diff
@@ -0,0 +1,32 @@
+params.read_triplets = "$projectDir/test_data/reads/reads_{1,2,3}*.gz"
+params.outdir = "results"
+
+process UMI_INJECTOR {
+    container 'umi-injector'
+    cpus 2
+
+    input:
+    tuple val(sample_id), path(read_triplets)
+
+    output:
+    tuple val(sample_id), path('*_with_umi.fastq.gz')
+
+    publishDir params.outdir
+    script:
+    """
+        umi-injector.sh --in1 ${read_triplets[0]} --umi ${read_triplets[1]}  --in2 ${read_triplets[2]} \
+            --out1 ${sample_id}_R1_with_umi.fastq.gz --out2 ${sample_id}_R2_with_umi.fastq.gz \
+            --threads ${task.cpus} --sep ':'
```

Comment:
I think I will keep imposing (de)compression for all 5 files for the Nextflow wrapper. So let's assume there are 5 pigz processes running to simplify the logic here. My gut feeling is to have at least one process-thread per cpu to not overbook the cluster. So if there are 8 cpus given, I would prefer having "--threads=2" rather than "--threads=1". Does `floor(task.cpus / 5) + 1)` work perhaps?

Reply:
Yes, this sounds good to me!

Reply:
Not really, no?

If you have e.g. `task.cpus=6`, you would get `floor(6/5)+1` is 2. So that would then actually use 10 threads for the pigz alone.

Reply:
Right I think I neglected the non-pigz processes...